test/Microsoft.AspNetCore.SignalR.Tests.Utils/LogSinkProvider.cs

```diff
@@ -0,0 +1,77 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Logging.Testing;
+
+namespace Microsoft.AspNetCore.SignalR.Tests
+{
+    // TestSink doesn't seem to be thread-safe :(.
```

Comment:
It should be now.

/cc @JunTaoLuo

Reply:
One thing that LogSinkProvider does that TestSink probably doesn't is raise an event when an item is added for it.

I can update the comment.

Would TestSink having an event make sense? It is just used internally by aspnet team right?